src/dbnode/persist/fs/retriever.go

```diff
@@ -299,12 +306,19 @@ func (r *blockRetriever) fetchBatch(
 
 	// Sort the requests by offset into the file before seeking
 	// to ensure all seeks are in ascending order
+	var entrySize int
 	for _, req := range reqs {
+		if err := r.opts.QueryStats().UpdateBytesRead(entrySize); err != nil {
+			req.onError(err)
+			continue
+		}
+
 		entry, err := seeker.SeekIndexEntry(req.id, seekerResources)
 		if err != nil && err != errSeekIDNotFound {
 			req.onError(err)
 			continue
 		}
+		entrySize = int(entry.Size)
```

Comment:
nit: better to do the update here? Otherwise end up in the case where we're always calling it with 0 (especially if the limit check fails)

Reply:
This was just so we could avoid even that first `SeekIndexEntry` call. But agreed probably is more readable to do after and rely on the `Exceeded` call before the loop